services/director-v2/src/simcore_service_director_v2/modules/db/repositories/internet_to_groups.py

```diff
@@ -0,0 +1,22 @@
+import sqlalchemy as sa
+from models_library.users import UserID
+
+from ..tables import internet_to_groups, user_to_groups
+from ._base import BaseRepository
+
+
+class InternetToGroupsRepository(BaseRepository):
+    async def has_access(self, user_id: UserID) -> bool:
+        async with self.db_engine.acquire() as conn:
+            # checks if one of the groups which the user is part of has internet access
+            select_stmt = sa.select([internet_to_groups.c.has_access]).select_from(
```

Comment:
This query checks all rows with has_access == True while you just need one single match, right?
Can you adapt it to something like:
```python
    async with engine.acquire() as conn:
        query = exists().where(mytable.columns.has_active == True)
        result = await conn.scalar(query)
```

Reply:
It's a good idea, but can't make it work. Will leave it as is for now.